src/Functions/neighbor.cpp

```diff
@@ -50,6 +50,8 @@ class FunctionNeighbor : public IFunction
 
     bool useDefaultImplementationForConstants() const override { return false; }
 
+    bool useDefaultImplementationForLowCardinalityColumns() const override { return false; }
```

Comment:
What is the point? Why it was working wrong?

Reply:
The optimization happened. Function was executed over dictionary

Reply:
Could you please add a comment?